File: 153-Find-Minimum-in-Rotated-Sorted-Array.py
class Solution(object):
    def findMin(self, nums):
        l = 0
        r = len(nums) - 1       
        while l < r:
            m = (l + r) // 2
            if nums[m] < nums[r]:
                r = m
            else:
                l = m + 1
        return nums[l]

File: neetcode/153-Find-Minimum-in-Rotated-Sorted-Array.ts
function findMin(nums: number[]): number {
  let l = 0, r = nums.length - 1

  while (l < r) {
    const m = Math.floor((r - l) / 2) + l

    if (nums[m] < nums[r]) r = m
    else l = m + 1
  }

  return nums[l]
}
